Core/Meta/Expression.cs

```diff
@@ -1,38 +1,39 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Numerics;
 using System.Text;
 using System.Threading.Tasks;
 using Core.Lexer.Tokenization.Models;
 
 namespace Core.Meta
 {
     public abstract record Expression(Span Span) {
-        public abstract long Value();
+        public abstract BigInteger Value();
```

Comment:
This is probably good also for the case where they input a really large integer, then our errors can be a little more clear.